include/swift/SIL/SILSuccessor.h

```diff
@@ -87,33 +87,45 @@ class SILSuccessor {
   class pred_iterator {
     SILSuccessor *Cur;
 
+    // Cache the basic block to avoid repeated pointer chasing.
+    SILBasicBlock *BB;
```

Comment:
Two things:

1. Can you use Block instead of BB for this name.
2. I am worried about having to call cacheBasicBlock everywhere. It is an extra step that someone may forget. At least I think it is important to have some sort of assert when you access the block to show that the block is the correct block, even though I would prefer if there was an algebraic way that the code could guarantee that BB will be changed if Cur is changed. It seems almost silly though to say wrap the SILSuccessor field in another struct that makes sure that BB is always updated.

Reply:
Sure, I can change the name.

I think if the maintainer of the iterator is comfortable with how C++ iterators work, then the calls to `cacheBasicBlock` will very limited in scope. That being said, if you don't like this tradeoff, then I'll just abandon the pull request. (I'm not invested in this pull request. It just felt like a "freebie" while I was hacking on some other stuff.)